### DISCUSSION TITLE
Searching across all volumes for a file?

### DISCUSSION THREAD

**Discussion by lonnietc** [seaweedfs/seaweedfs#2419] [Q&A]
Hello,

I have been playing around with Seaweed and working with the information on the Github site, but was wondering if there was a way to search for a file across all volumes?

Thanks

**Comment by chrislusf** [comment]
What's your use case? Search by name, attributes, or text?

**Comment by lonnietc** [comment]
Hello,

Mostly exploring by name.

Considering how it might be used in a new P2P type application where each
node is a Seaweed volume.

**Comment by chrislusf** [comment]
You can have this process running, and index SeaweedFS metadata changes on an elastic search server:
`weed filer.meta.tail -timeAgo=30h -es=http://<elasticSearchServerHost>:<port> -es.index=seaweedfs`

I do not know how well it works. Let me know, so we can add more wiki on this.

**Comment by lonnietc** [comment]
Thanks for the information as it will be very useful in this build effort.

Have done some basic testing with Seaweed and it really is an awesome piece
of software.
